goblin/driver/cluster.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2016 ZEROFAIL
+#
+# This file is part of Goblin.
+#
+# Goblin is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Goblin is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Goblin.  If not, see <http://www.gnu.org/licenses/>.
+
+import asyncio
+import collections
+import configparser
+import json
+import ssl
+
+import yaml
+
+from goblin import driver, exception
+
+
+def my_import(name):
+    components = name.split('.')
+    mod = __import__(components[0])
+    for comp in components[1:]:
+        mod = getattr(mod, comp)
+    return mod
+
+
+class Cluster:
+    """
+    A cluster of Gremlin Server hosts. This object provides the main high
+    level interface used by the :py:mod:`goblin.driver` module.
+
+    :param asyncio.BaseEventLoop loop:
+    """
+
+    DEFAULT_CONFIG = {
+        'scheme': 'ws',
+        'hosts': ['localhost'],
+        'port': 8182,
```

Comment:
I just realized gremlin-console's `remote.yaml` works this way too -- all hosts must use the same port. Seems odd to me, but I guess it's alright.